client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -117,27 +117,31 @@ private void closeConnection(Throwable exceptionToInflightRequests) {
             request.completeExceptionally(exceptionToInflightRequests);
         }
     }
-    
+
     public <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(long requestId, T request) {
         return connection.thenCompose(c -> {
             log.debug("Sending request: {}", request);
             CompletableFuture<Reply> reply = new CompletableFuture<>();
             synchronized (lock) {
                 requests.put(requestId, reply);
             }
-            try {
-                c.send(request);
```

Comment:
This is actually a behaviour change, because before this change the sendRequest returned a CompletableFuture only "after" sending the request on the wire (c.send is blocking), now we are returning the CompletableFuture as soon as possible.
This in theory should not be a problem.

I can try to change this call by "composing" with a wait for the completion of the sendAsync.

In my new change the returned CompletableFuture will *complete* only "after" the command has been sent on the wire.
In theory the caller of this method  should not matter about this causality because on the wire we can always have delays.
I wonder if the system test which is failing is buggy and relies on some fortunate sequences of events.

what you you think @tkaitchuck @shrids ?

Reply:
It's worth trying. But I don't see how this would affect the append path...